### PR TITLE
Make pn.json filename configurable, other minor changes

### PR DESCRIPTION
**Pull request by hansgschossmann**
Make pn.json filename configurable to allow local sharing with other containers.
Other minor changes and doc updates.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,7 +110,8 @@ docker run --rm -it -p 50000:50000 -p 8080:8080 --name opcplc mcr.microsoft.com/
 ~~~
 
 ## OPC Publisher file (pn.json)
-The option `sph` shows and dumps a pn.json file that matches the configuration. In addition, a web server hosts the file on a configurable port (`wp`, default 8080): e.g. http://localhost:8080/pn.json
+The options `sph` and `sp` show and dump an OPC Publisher configuration file (default name: `pn.json`) that matches the configuration. In addition, a web server hosts the file on a configurable port (`wp`, default 8080): e.g. http://localhost:8080/pn.json
+Additionally you are can set the configuration file name via the option `spf`.
```
Comment (luiscantero) [inline README.md:114]: Additionally you are can set -> Additionally, you can set

File: src/PlcSimulation.cs [hunk 2]
```diff
@@ -120,7 +99,7 @@ private void SpikeGenerator(object state)
             {
                 // todo calculate
                 nextValue = SimulationMaxAmplitude * 10;
-                Logger.Verbose($"generate spike anomaly");
+                Logger.Verbose("generate spike anomaly");
```
Comment (luiscantero) [inline src/PlcSimulation.cs:102]: Generate

File: src/PlcSimulation.cs [hunk 3]
```diff
@@ -149,7 +128,7 @@ private void DipGenerator(object state)
             if (GenerateDips && _dipCycleInPhase == _dipAnomalyCycle)
             {
                 nextValue = SimulationMaxAmplitude * -10;
-                Logger.Verbose($"generate dip anomaly");
+                Logger.Verbose("generate dip anomaly");
```
Comment (luiscantero) [inline src/PlcSimulation.cs:131]: Generate

File: src/PlcSimulation.cs [hunk 4]
```diff
@@ -177,8 +156,8 @@ private void PosTrendGenerator(object state)
             double nextValue = TREND_BASEVALUE;
             if (GeneratePosTrend && _posTrendPhase >= _posTrendAnomalyPhase)
             {
-                nextValue = TREND_BASEVALUE + (_posTrendPhase - _posTrendAnomalyPhase) / 10;
-                Logger.Verbose($"generate postrend anomaly");
+                nextValue = TREND_BASEVALUE + ((_posTrendPhase - _posTrendAnomalyPhase) / 10);
+                Logger.Verbose("generate postrend anomaly");
```
Comment (luiscantero) [inline src/PlcSimulation.cs:160]: Generate

File: src/PlcSimulation.cs [hunk 5]
```diff
@@ -201,8 +180,8 @@ private void NegTrendGenerator(object state)
             double nextValue = TREND_BASEVALUE;
             if (GenerateNegTrend && _negTrendPhase >= _negTrendAnomalyPhase)
             {
-                nextValue = TREND_BASEVALUE - (_negTrendPhase - _negTrendAnomalyPhase) / 10;
-                Logger.Verbose($"generate negtrend anomaly");
+                nextValue = TREND_BASEVALUE - ((_negTrendPhase - _negTrendAnomalyPhase) / 10);
+                Logger.Verbose("generate negtrend anomaly");
```
Comment (luiscantero) [inline src/PlcSimulation.cs:184]: Generate

File: src/Startup.cs [hunk 6]
```diff
@@ -33,7 +37,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             // Serve pn.json
             app.Run(async context =>
             {
-                if (context.Request.Path == $"/{Program.PnJson}" &&
+                if (context.Request.Method == "GET" && context.Request.Path == $"{Program.PnJson}" &&
```
Comment (luiscantero) [inline src/Startup.cs:40]: Does this work without forward slash as prefix? $"{Program.PnJson}"
If so change: $"{Program.PnJson}" -> Program.PnJson

**Review by luiscantero — CHANGES_REQUESTED** [review]

File: src/Startup.cs (hunk 6 above)
Comment (hansgschossmann) [reply src/Startup.cs]: good catch. for the default value i had to add a /.

File: src/Properties/launchSettings.json [hunk 7]
```diff
@@ -1,7 +1,8 @@
 {
   "profiles": {
     "opc-plc": {
-      "commandName": "Project"
+      "commandName": "Project",
+      "commandLineArgs": "--aa --sph --fn 10 "
```
Comment (luiscantero) [inline src/Properties/launchSettings.json:5]: Did you intend to check-in this file?

**Review by luiscantero — APPROVED** [review]

File: src/Properties/launchSettings.json (hunk 7 above)
Comment (hansgschossmann) [reply src/Properties/launchSettings.json:5]: actually it is already checked in. i added it to .gitignore in this checkin. let me try to remove i already with this PR from the repo.

**Pull request merged**